doc/src/manual/methods.md

```diff
@@ -367,9 +364,9 @@ Stacktrace:
 [...]
 ```
 
-As you can see, the type of the appended element must match the element type of the vector it
-is appended to, or else a [`MethodError`](@ref) is raised. In the following example, the method type parameter
-`T` is used as the return value:
+The type parameter `T` in the signature of the function ensures that appending element `x` to `v` preserves the type of the new `v`. In the function signature, `[v..., x]`, the three dots `...` stand for the regular splatting operator, while the square brackets stand for the vector with all the elements of `v` unpacked and `x` appended to it. The meaning of `{T} = [v..., x]` is that all the elements of the vector are constrained to be of the same type. Note that `{T} = v` has a different meaning as `v` refers to the container and not its elements. Note also that `{T} = [v...]` would not quite work as intended because the method constructor does not yet know that `x` is to become an element of the new vector `v`.
```

Comment:
Talking about `{T} = [v..., x]` like this does not make much sense. `{T}` is part of the method signature, not the body. As such, `{T} = [v..., x]` doesn’t have a meaning, it’s not even valid Julia syntax (you cannot assign a value to `{T}`).

Reply:
Yes, I realize that. And this is what made the whole `{T} = ...` difficult to process. Perhaps giving more examples would help. What would the syntax for a `Tuplet`? (square brackets too?)  When can the curly brackets around `T` be omitted? Please do make suggestions that could be used to improve the docs. Thanks.

Reply:
Maybe it’s easier to talk about it when using the long-form method syntax. The following are equivalent:
```julia
myappend(v::Vector{T}, x::T) where {T} = [v..., x]
```
```julia
function myappend(v::Vector{T}, x::T) where {T}
    return [v..., x]
end
```
As you can see, the second version has no `{T} = ...`.

To return a `Tuple` instead of a `Vector`, you would use round brackets, i.e., `return (v..., x)`. However, this has nothing to do with the parametric method syntax that is explained in this section. It is just the usual syntax to create `Vector`s and `Tuple`s, respectively.

The curly brackets around `T` can be omitted here, they are only necessary when using several type parameters in the signature. For example, one could define a method `myappend(v::Vector{T}, x::S) where {T,S}`. Here, the curly brackets around `T,S` are necessary.

Reply:
Thanks sostock! I see I had totally misunderstood the function signature syntax. I thought the ``= [v...,x]`` part was part of the signature, but now I see that it's only the ``where {T}`` part. 

Introducing the long-form method syntax first, and then saying that it could be written in an equivalent short-form, as you do, would have been clear. I'm not sure how many new users are likely to be confused by this. Perhaps more than one, because ``where {T}`` is pretty cryptic (I've got to re-read the docs to see if I understand it).

Reply:
> Talking about `{T} = [v..., x]` like this does not make much sense. `{T}` is part of the method signature, not the body. As such, `{T} = [v..., x]` doesn’t have a meaning, it’s not even valid Julia syntax (you cannot assign a value to `{T}`).

After seeing the long-form version, your comment now makes sense. I knew {T} was part of the method signature, but somehow I ended up thinking that the part after the equal sign was also a part of it. I was confused by the equal sign.  I think it really helps to show the long-form and short-form side by side. The line break in the long form makes it clear where the signature ends, the equal sign not so much (to a newbie anyway). Thanks again for your help.